test/repository_test.rb

```diff
@@ -299,4 +299,61 @@ def to_int
       end
     end
   end
+
+  describe "with sql adapter" do
+    before do
+      UserRepository.adapter    = Lotus::Model::Adapters::SqlAdapter.new(MAPPER, SQLITE_CONNECTION_STRING)
+      ArticleRepository.adapter = Lotus::Model::Adapters::SqlAdapter.new(MAPPER, SQLITE_CONNECTION_STRING)
+
+      UserRepository.collection    = :users
+      ArticleRepository.collection = :articles
+
+      UserRepository.clear
+      ArticleRepository.clear
+
+      ArticleRepository.create(article1)
+    end
+
+    describe '.transaction' do
+      it 'if an exception is raised the size of articles is equal to 1' do
+        ArticleRepository.all.size.must_equal 1
+        exception = -> { ArticleRepository.transaction do
+          ArticleRepository.create(article2)
+          raise Exception
+        end }
+        exception.must_raise Exception
+        ArticleRepository.all.size.must_equal 1
+      end
+
+      it "if an exception isn't raised the size of articles is equal to 2" do
+        ArticleRepository.all.size.must_equal 1
+        ArticleRepository.transaction do
+          ArticleRepository.create(article2)
+        end
+        ArticleRepository.all.size.must_equal 2
+      end
+    end
+  end
+
+  describe "with memory adapter" do
```

Comment:
From user perspective, I think it's worth to raise something a meaningful message that the adapter does not support transaction rather than a generic RuntimeError. But that could be another PR.